src/Features/Core/Portable/PullMemberUp/Dialog/SymbolDependentsBuilder.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.FindSymbols;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog
+{
+    internal class SymbolDependentsBuilder : SyntaxWalker
+    {
+        internal List<ISymbol> SymbolDependentsList { get; }
+
+        private SemanticModel SemanticModel { get; set; }
+
+        private HashSet<ISymbol> SymbolSet { get; set; }
+
+        private Document ContextDocument { get; set; }
+
+        private CancellationToken CancellationToken { get; set; }
+
+        private SymbolDependentsBuilder()
+        {
+            SymbolDependentsList = new List<ISymbol>();
+        }
+
+        internal static List<ISymbol> Build(
+            SemanticModel semanticModel,
+            ISymbol userSelectedNodeSymbol,
+            HashSet<ISymbol> members,
+            Document contextDocument,
+            CancellationToken cancellationToken)
+        {
+            var builder = new SymbolDependentsBuilder()
+            {
+                SymbolSet = new HashSet<ISymbol>(members),
+                SemanticModel = semanticModel,
+                ContextDocument = contextDocument,
+                CancellationToken = cancellationToken
+            };
+
+            var selectedSyntax = userSelectedNodeSymbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
```

Comment:
1. when are you doing anything when DeclaringSyntaxReferences is empty?
2. why are we only looking at one of the references, and not all of them?